java/src/jni/h5sImp.c

```diff
@@ -1340,6 +1340,389 @@ Java_hdf_hdf5lib_H5_H5Sget_1regular_1hyperslab(JNIEnv *env, jclass clss, jlong s
     return;
 } /* end Java_hdf_hdf5lib_H5_H5Sget_1regular_1hyperslab */
 
+/*
+ * Class:     hdf_hdf5lib_H5
+ * Method:    H5Sselect_copy
+ * Signature: (JJ)V
+ */
+JNIEXPORT void JNICALL
+Java_hdf_hdf5lib_H5_H5Sselect_1copy(JNIEnv *env, jclass clss, jlong dst_id, jlong src_id)
+{
+    hssize_t retVal = -1;
+
+    UNUSED(clss);
+
+    if ((retVal = H5Sselect_copy((hid_t)dst_id, (hid_t)src_id)) < 0)
+        H5_LIBRARY_ERROR(ENVONLY);
+
+done:
+    return;
+} /* end Java_hdf_hdf5lib_H5_H5Sselect_1copy */
+
+/*
+ * Class:     hdf_hdf5lib_H5
+ * Method:    H5Sselect_shape_same
+ * Signature: (JJ)Z
+ */
+JNIEXPORT jboolean JNICALL
+Java_hdf_hdf5lib_H5_H5Sselect_1shape_1same(JNIEnv *env, jclass clss, jlong space1_id, jlong space2_id)
+{
+    htri_t bval = JNI_FALSE;
+
+    UNUSED(clss);
+
+    if ((bval = H5Sselect_shape_same((hid_t)space1_id, (hid_t)space2_id)) < 0)
+        H5_LIBRARY_ERROR(ENVONLY);
+
+    bval = (bval > 0) ? JNI_TRUE : JNI_FALSE;
+
+done:
+    return (jboolean)bval;
+} /* end Java_hdf_hdf5lib_H5_H5Sselect_1shape_1same */
+
+/*
+ * Class:     hdf_hdf5lib_H5
+ * Method:    H5Sselect_adjust
+ * Signature: (J[J)V
+ */
+JNIEXPORT void JNICALL
+Java_hdf_hdf5lib_H5_H5Sselect_1adjust(JNIEnv *env, jclass clss, jlong space_id, jlongArray offset)
```

Comment:
H5Sselect_adjust can return an error, so probably shouldn't return void.

Reply:
S.B. handled by        H5_LIBRARY_ERROR(ENVONLY);

Reply:
Like the other case, I don't think letting H5_LIBRARY_ERROR handle it is the normal way we do things. I believe that's only going to throw an error stack. From the API perspective I think that means there isn't a reasonable way to check that this failed. You just have to catch an exception which isn't really clean.